src/navigation/RouterConfig.tsx

```diff
@@ -63,7 +63,9 @@ export const RouterConfig = ({ userType }: { user: any; userType: string }) => {
           break;
         case AgreementState.PendingAgreement:
         case AgreementState.ApprovedAgreement:
-          history.push(DASHBOARD);
+          if (location.pathname === "/") {
```

Comment:
What is the behavior of that condition, why is it needed?

Reply:
This Effect triggers at page load. The previous behaviour was that if the user reloads the page, react router correctly renders the same route, but this effect redirects always on reload to dashboard route, i think this behaviour is accidental. This condition now redirects to dashboard only if we are the root route of the domain

Reply:
this effect prevent hor reloading to work properly also